src/components/ClassList.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState } from "react";
+import {
+    Box,
+    Flex,
+    Badge,
+    Center,
+    AspectRatio,
+    Spacer,
+    Image,
+    List,
+    ListItem,
+    VStack,
+    Grid,
+    GridItem,
+} from "@chakra-ui/react";
+import type { ClassCardInfo } from "models/Class";
+export const ClassList: React.FC<{ classInfo: ClassCardInfo[] }> = ({
+    classInfo,
+}) => {
+    return (
+        <Center width="100%">
+            <List spacing="5" width="100%">
+                {classInfo.map((item, idx) => {
+                    return (
+                        <ListItem
+                            borderColor="gray.200"
+                            _hover={{ borderColor: "gray.600" }}
+                            borderWidth={2}
+                        >
+                            <Grid templateColumns="repeat(5, 1fr)" gap={6}>
+                                <GridItem>
+                                    <AspectRatio width="100%" ratio={1}>
+                                        <Image
+                                            src={item.image}
+                                            fit="cover"
+                                            alt={item.name}
+                                        />
+                                    </AspectRatio>
+                                </GridItem>
+                                <GridItem colSpan={3}>
+                                    <VStack
+                                        align="left"
+                                        justify="center"
+                                        height="100%"
+                                    >
+                                        <Box fontWeight="bold" as="h2">
+                                            {item.name}
+                                        </Box>
+                                        <Box
+                                            as="span"
+                                            color="gray.600"
+                                            fontSize="sm"
+                                            textTransform="capitalize"
+                                        >
+                                            {item.weekday
```

Comment:
Just a suggestion but maybe we can make this a function since we might use it elsewhere too? Also would improve code readability.